Formula/podman.rb

```diff
@@ -17,24 +17,45 @@ class Podman < Formula
   depends_on "go-md2man" => :build
   depends_on "qemu" if Hardware::CPU.intel?
 
+  resource "gvproxy" do
+    url "https://github.com/containers/gvisor-tap-vsock/archive/v0.1.0.tar.gz"
+    sha256 "e1e1bec2fc42039da1ae68d382d4560a27c04bbe2aae535837294dd6773e88e0"
+  end
+
   def install
-    system "make", "podman-remote-darwin"
-    bin.install "bin/darwin/podman"
+    os = "darwin"
+    on_linux do
+      os = "linux"
+    end
```

Comment:
```suggestion
    os = if OS.mac?
      "darwin"
    else
      "linux"
    end
```

apologies if this contradicts what others have said but @Homebrew/maintainers I think this is an example of a case where the if/else form is dramatically easier to read. Not sure if an audit/style check forbids this but: if it does, we should make it not apply for `def install` (and not let that block this PR).

Reply:
This pattern actually appears all over the place. It's part of what motivated me to open Homebrew/brew#11687 but neglected to include it when I was writing things down 😅

Reply:
Yeh, ouch. Would be good to improve this but definitely not blocking this PR then.

Reply:
Mixing those forms brings some confusion and lack of consistency. I don't think we should be using both interchangeably, the readability is in my opinion on par in both cases. If we start using `OS.mac?` again here and there, then what was the point in the first place changing all those occurrences earlier. I would say we stay with `on_os` blocks exclusively.

Reply:
I think the `if` form is significantly better:
1. it's idiomatic Ruby, which means one doesn't need to be a regular Homebrew/core contributor to understand at a glance what's going on
2. `on_os` blocks also have slightly more complicated variable scoping rules

Reply:
This one is so common we could even have a `OS.uname`/`kernel_name`.

Reply:
If so then I would say drop `on_os` blocks altogether and just use `if` statements like before and everywhere else. Then the revisions problem would automatically go away cause we could just migrate those too and that's it.

Reply:
See Homebrew/brew#11687 for a discussion of why we still want `on_os` blocks even if we're not using them in `install` methods.

Reply:
> If so then I would say drop `on_os` blocks altogether and just use `if` statements like before and everywhere else.

I'd be fine with doing that in `install`/`post_install`/`test` methods but definitely not in the class definition.